pkg/storaged/side-panel.jsx

```diff
@@ -32,12 +32,28 @@ export class SidePanel extends React.Component {
     constructor() {
         super();
         this.state = { collapsed: true };
+        this.current_rows_keys = [];
+        this.new_keys = [];
     }
 
     render() {
         let show_all_button = null;
-        let rows = this.props.rows;
+        let rows = this.props.rows.filter(row => !!row);
+
+        // Find new items for animations
+        const current_keys = [];
+        rows.forEach(row => { current_keys.push(row.key) });
+
+        if (JSON.stringify(this.current_rows_keys) !== JSON.stringify(current_keys)) {
+            if (this.current_rows_keys.length !== 0) {
+                const new_keys = current_keys.filter(key => this.current_rows_keys.indexOf(key) === -1);
+                if (new_keys.length)
+                    this.new_keys.push(...new_keys);
```

Comment:
This looks strange -- pushing a copy of an array onto itself? That gets dangerously close to [Russel's pararox](https://en.wikipedia.org/wiki/Russell%27s_paradox) and hence making the universe explode (or implode.. I always forget which :grinning: )

Joke aside, you mean something like "concat" here (except that this is not an inline modification). Can't you just do `this.new_keys = new_keys` here?

Reply:
Illustration, in case it's not clear:

```node
> a = []
[]
> b = [1,2]
[ 1, 2 ]
> a.push(b)
1
> a
[ [ 1, 2 ] ]
> 
```

Reply:
> Joke aside, you mean something like "concat" here (except that this is not an inline modification). Can't you just do this.new_keys = new_keys here?

Same as for ListingTable: https://github.com/cockpit-project/cockpit/pull/17094#discussion_r826726870

I want to keep list of all items that were added after initial load. For that I keep global state of new items and only append into it. Otherwise I could just have local array of new keys for each render, but that might cause issues when more items are added one by one in quick succession

Reply:
> Illustration, in case it's not clear:

Not sure what it demonstrates. I need the spread operator exactly to avoid creating array of arrays as that is hell then to find element in.

Reply:
Oh, I wasn't aware of *that* usage of the spread operator. Your linked comment used `[...var]`, which is by and large "a copy of `var`", but this doesn't have the `[]` and thus is more like calling .push() with lots of arguments. That works fine (I just found it hard to read, but now I know it).